src/DIRAC/Core/Utilities/DIRACScript.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+__RCSID__ = "$Id$"
+
+from collections import defaultdict
+import functools
+
+import six
+
+from DIRAC import gLogger
+
+
+class DIRACScript(object):
+  """Decorator for providing command line executables
+
+  All console-scripts entrypoints in DIRAC and downstream extensions should be
+  wrapped in this decorator to allow extensions to override any entry_point.
+  """
+  def __init__(self):
```

Comment:
This constructor could be useful for allowing something like below. It's not strictly needed though so it can be deferred to another release if we want (or never). I'll wait for opinions before implementing it.

```python
@DiracScript(
    switches=[
        ("f:", "file=", "File to use as user key"),
        ("i", "version", "Print version"),
        ("n", "novoms", "Disable VOMS"),
        ("v", "checkvalid", "Return error if the proxy is invalid"),
        ("x", "nocs", "Disable CS"),
        ("e", "steps", "Show steps info"),
        ("j", "noclockcheck", "Disable checking if time is ok"),
        ("m", "uploadedinfo", "Show uploaded proxies info"),
    ],
    requriesProxy=False,
)
def proxy_info(*args, **kwargs):
```

Reply:
I would imagine `requriesProxy=True` would be the default if we did this.

>  For the switches, maybe we could maybe look at something like click ? Anyway, I would differ it to later

I think the current parsing is too inconsistent to repliment using `click` so while it would be nice we need to be prepared to a single big backwards-incompatible break or, more likely, a transisition to a new CLI.

> Anyway, I would differ it to later

In hindsight I agree that it's best left out of this PR to ensure the diff is reviewable.